pkg/devfile/parser/data/v2/commands.go

```diff
@@ -65,13 +64,21 @@ func (d *DevfileV2) AddCommands(commands []v1.Command) error {
 }
 
 // UpdateCommand updates the command with the given id
-func (d *DevfileV2) UpdateCommand(command v1.Command) {
+// return an error if the command is not found
+func (d *DevfileV2) UpdateCommand(command v1.Command) error {
+	found := false
 	for i := range d.Commands {
-		if strings.ToLower(d.Commands[i].Id) == strings.ToLower(command.Id) {
+		if d.Commands[i].Id == command.Id {
+			found = true
 			d.Commands[i] = command
-			d.Commands[i].Id = strings.ToLower(d.Commands[i].Id)
+			d.Commands[i].Id = d.Commands[i].Id
+			break
 		}
 	}
+	if !found {
+		return fmt.Errorf("update command failed: command %s not found", command.Id)
+	}
+	return nil
```

Comment:
if we return instead of break on a successful update, we can remove the flag and this check and return an err at the end, we can make the same updates for the rest of the other update funcs

Reply:
updated